agent/main.go

```diff
@@ -148,7 +148,10 @@ func main() {
 	// create handlers
 	eventHandler := handler.NewEvent(k8sClient, apiClient)
 	workloadHandler := handler.NewWorkload(k8sClient, apiClient)
-	linkerdInfoHandler := handler.NewLinkerdInfo(k8sClient, apiClient)
+
+	diagnostics := make(chan handler.DiagnosticsDataRequest, 1)
+	linkerdInfoHandler := handler.NewLinkerdInfo(k8sClient, apiClient, diagnostics)
+	manageAgentHandler := handler.NewManageAgent(apiClient, diagnostics)
```

Comment:
You are absolutely right. Looking at the protobuf definition it not clear what actually happens when you send a message to the `ManageAgent`. Even worse, my line of thinking was that we can keep doing that for future functionality. Namely have a new command on `ManageAgent` that can do one of a few things: 

- send no response at all
- send a response to the LinkerdInfo endpoint
- send a response to any other endpoint that we do not have at the moment but we might have in the future.

This does not feel like a good design, I was mostly governed by the feedback I got on the notion doc but now I realize it makes more sense from API design perspective to have things be a bit more principled. 

So should we have a bi directional stream instead that multiplexes all agent related communication where we send a command and expect to receive a response as a result of this command? Would that make sense. At least then the protobuf api will more precisely capture the communication patterns that are possible

Reply:
I don't think that we should use a bidirectional stream that both sends commands to the agent and receives the results of running those commands. My thinking here is that some commands may not need responses (e.g adjusting the proxy log level), whereas other commands may send us lots of data. If we use a bidirectional stream then we can't balance the responses across all of our API servers. It would be better to send the command on one stream, and receive the data via separate unary RPCs (or streaming RPCs as needed).

If we're worried about maintaining a more principled relationship between commands and responses, I suggest that we address that with consistent naming across our API definition. For example, we could provide a `ProxyDiangostics` endpoint that mirrors the naming of the `GetProxyDiagnostics` command that we're adding here. That said, I don't really mind that the current implementation is calling the `LinkerdInfo` endpoint instead.

Reply:
> If we're worried about maintaining a more principled relationship between commands and responses, I suggest that we address that with consistent naming across our API definition. For example, we could provide a ProxyDiangostics endpoint that mirrors the naming of the GetProxyDiagnostics command that we're adding here. That said, I don't really mind that the current implementation is calling the LinkerdInfo endpoint instead.

Yes, we already covered that in our Friday conversation and I have reflected that in the design doc, so this is how we proceed